phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/server/QueryServer.java

```diff
@@ -274,6 +282,47 @@ public int run(String[] args) throws Exception {
   }
 
   /**
+   * Use the correctly way to extract end user.
+   */
+
+  static class PhoenixRemoteUserExtractor implements RemoteUserExtractor{
+    private final HttpQueryStringParameterRemoteUserExtractor paramRemoteUserExtractor;
+    private final HttpRequestRemoteUserExtractor requestRemoteUserExtractor;
+    private final boolean enableDoAs;
+    private final String doAsParam;
+
+    public PhoenixRemoteUserExtractor(Configuration conf) {
+      this.requestRemoteUserExtractor = new HttpRequestRemoteUserExtractor();
+      this.doAsParam = conf.get(QueryServices.QUERY_SERVER_DOAS_PARAM,
+              QueryServicesOptions.DEFAULT_QUERY_SERVER_DOAS_PARAM);
+      this.paramRemoteUserExtractor = new HttpQueryStringParameterRemoteUserExtractor(doAsParam);
+      this.enableDoAs = conf.getBoolean(QueryServices.QUERY_SERVER_DOAS_ENABLED_ATTRIB,
```

Comment:
Can you move this check of whether or not we enable `doAs` above to selectively call `withRemoteUserExtractor`, please?